storage/src/vespa/storage/persistence/filestorage/filestormanager.cpp

```diff
@@ -127,6 +130,14 @@ FileStorManager::createRegisteredHandler(ServiceLayerComponent & component)
                                                  *_bucketOwnershipNotifier, *_metrics->disks[0]->threads[index]));
     return *_persistenceHandlers.back();
 }
+
+PersistenceHandler &
+FileStorManager::getThreadLocalHandler() {
+    if (_G_threadLocalHandler == nullptr) {
+        _G_threadLocalHandler = & createRegisteredHandler(_component);
```

Comment:
How do we ensure the `_persistenceHandlers.size() < _metrics->disks[0]->threads.size()` invariant holds when an "arbitrary" number of handlers can be registered?

Reply:
The assumption is that this code will only be invoked from network threads and response threads.
Those are accounted for in the last commit.